spec.html

```diff
@@ -42955,6 +42972,392 @@ <h1>Properties of _TypedArray_ Instances</h1>
+            1. Set _index_ to _index_ + 1.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-skipasciiwhitespace" type="abstract operation">
+        <h1>
+          SkipAsciiWhitespace (
+            _string_: a String,
+            _index_: a non-negative integer,
+          ): a non-negative integer
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Let _length_ be the length of _string_.
+          1. Repeat, while _index_ &lt; _length_,
+            1. Let _char_ be the code unit at index _index_ of _string_.
+            1. If _char_ is not one of 0x0009 (TAB), 0x000A (LF), 0x000C (FF), 0x000D (CR), or 0x0020 (SPACE), then
+              1. Return _index_.
+            1. Set _index_ to _index_ + 1.
+          1. Return _index_.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-decodefinalbase64chunk" type="abstract operation">
+        <h1>
+          DecodeFinalBase64Chunk (
+            _chunk_: a String of length 2 or 3,
+            _throwOnExtraBits_: a Boolean,
+          ): either a normal completion containing a List of byte values, or a throw completion
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. Let _chunkLength_ be the length of _chunk_.
+          1. If _chunkLength_ = 2, then
+            1. Set _chunk_ to the string-concatenation of _chunk_ and *"AA"*.
+          1. Else,
+            1. Assert: _chunkLength_ is 3.
+            1. Set _chunk_ to the string-concatenation of _chunk_ and *"A"*.
+          1. Let _bytes_ be DecodeFullLengthBase64Chunk(_chunk_).
+          1. If _chunkLength_ = 2, then
+            1. If _throwOnExtraBits_ is *true* and _bytes_[1] ≠ 0, then
+              1. Throw a *SyntaxError* exception.
+            1. Return « _bytes_[0] ».
+          1. Else,
+            1. If _throwOnExtraBits_ is *true* and _bytes_[2] ≠ 0, then
+              1. Throw a *SyntaxError* exception.
+            1. Return « _bytes_[0], _bytes_[1] ».
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-decodefulllengthbase64chunk" type="abstract operation">
+        <h1>
+          DecodeFullLengthBase64Chunk (
+            _chunk_: a String of length 4,
+          ): a List of byte values of length 3
+        </h1>
+        <dl class="header">
+        </dl>
+        <p>The <dfn id="standard-base64-alphabet">standard base64 alphabet</dfn> is *"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/"*, i.e., the String whose elements are the code units corresponding to every letter and number in the Unicode Basic Latin block along with *"+"* and *"/"*.</p>
+        <emu-alg>
+          1. Let _byteSequence_ be the unique sequence of 3 bytes resulting from decoding _chunk_ as base64 (such that applying the base64 encoding specified in section 4 of <a href="https://datatracker.ietf.org/doc/html/rfc4648">RFC 4648</a> to _byteSequence_ would produce _chunk_).
```

Comment:
Normally, the spec uses "such that" to express a constraint, but that's presumably not what's intended here. Maybe:
```suggestion
          1. Let _byteSequence_ be the unique sequence of 3 bytes resulting from decoding _chunk_ as base64. (That is, applying the base64 encoding specified in section 4 of <a href="https://datatracker.ietf.org/doc/html/rfc4648">RFC 4648</a> to _byteSequence_ would produce _chunk_.)
```
I also tried the parenthetical as a NOTE step and as an Assert step. They work, but they don't have the same feel.

Alternatively, don't get into "decode" at all:
```suggestion
          1. Let _byteSequence_ be the unique sequence of 3 bytes such that applying the base64 encoding specified in section 4 of <a href="https://datatracker.ietf.org/doc/html/rfc4648">RFC 4648</a> to _byteSequence_ would produce _chunk_.
```

Reply:
I went with "i.e., the sequence such that", which makes it clearly expressing a constraint but I think reads better than either of these suggestions. In particular I do want to continue to describe this as "decoding" because that is how everyone thinks of it, even though the RFC does not formally define a decoding algorithm.